src/network/webtask.h

```diff
@@ -150,6 +150,8 @@ class WebTask : public NetworkTask {
         Idle,
         // Pending state
         Pending,
+        // Looping state
```

Comment:
This new state does not belong to the low-level network task. `WebTask` has no notion of retries and doesn't need this state for implementing its internal state machine.

Reply:
This does NOT control retries and the state IS used in WebTask itself. 

It controls the new state where the WebTask is in the idle loop between two requests. Before there was no need for this state because the new request was issued just after the old one. So the solution looks like a valid one.

I am not saying this is the only possible solution, so if you like to see another solution for this Bugfix. Please give some hints how.

Reply:
A task models the lifecycle of a single request. It could be restarted while not pending. Retries have to be controlled by the calling code.

Introducing a Trojan Horse into the base class just because some derived class wants to do implicit retries is the wrong approach.

I strongly recommend not to merge this PR. It introduces unneeded complexity.

Reply:
> A task models the lifecycle of a single request.

Sorry, that is not correct. You may have designed it like that original, but since MusicBrainzRecordingsTask that inherits from WebTask works with a couple of request this no longer the case. See: MusicBrainzRecordingsTask::continueWithNextRequest();

> Retries have to be controlled by the calling code.

Again, this change has nothing to do with reties.  

This PR is a valid solution for a real issue, maybe not perfect. So it would be helpful to suggest improvements instead of to title it with striking words.

Reply:
Sorry, I overlooked that. Long ago. I was confused by the wording "Looping" which is used out of context and has not meaning in the context of `WebTask`.

`continueWithNextRequest()` is undefined, probably a left over, please delete it.

I suggest to use a local timer in MusicBrainzRecordingsTask and invoke the next slotStart() with a delay. You could even override slotStart()/slotAbort() locally to detect a pending timer if needed.

Reply:
> I suggest to use a local timer in MusicBrainzRecordingsTask and invoke the next slotStart() with a delay. 

Done 

> You could even override slotStart()/slotAbort() locally to detect a pending timer if needed.

We have discussed that but rejected the idea because it either introduces code duplication or is a bugger refactoring. 
The additional state is a good solution because the Task is actually in that state.

Reply:
I still disagree. It's a move in the wrong direction.

Reply:
Which code duplication? It would just be a shallow wrapper around the base class slots that handles the pending time.

Reply:
If you really want to implement a throttling between subsequent requests then you could implement it entirely in the base class by adding a configuration parameter. But I don't recommend it, because the base class with its state machine is already complex.

The distributed approach introduced by this PR is hard to maintain and not needed.

Reply:
Which code duplication?

slotStart() and slotAbort()

> It is actually an supervision or overlay state and reflects the target design for a future refactoring

The additional state is a much more clean and maintainable solution. Less code duplication and a clear distinguished state from the other states that actually exists. I am pretty confident with the current solution and do not plan a future refactoring.

Reply:
I wish you luck.